pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -213,10 +213,15 @@ func GenerateTaskSpec(
 		}
 	}
 
+	if buildRun.Spec.Output == nil {
+		buildRun.Spec.Output = &buildv1alpha1.Image{}
+	}
```

Comment:
Let's not mutate the spec here.

```suggestion
	buildRunOutput := buildRun.Spec.Output
	if buildRunOutput == nil {
		buildRunOutput = &buildv1alpha1.Image{}
	}
```